packages/product-react/src/components/Input/Input.tsx

```diff
@@ -0,0 +1,112 @@
+import clsx from 'clsx';
+import React from 'react';
+import FieldWrapper from '../Common/FieldWrapper';
+import FormItemWrapper from '../Common/FormItemWrapper';
+import IconWrapper from '../Common/IconWrapper';
+import RTLWrapper from '../Common/RTLWrapper';
+
+export enum IconPosition {
+  Prepend = 'PREPEND',
+  End = 'END',
+  Start = 'START'
+}
+
+type InputProps = React.HTMLProps<HTMLInputElement> & {
+  active?: boolean;
+  compact?: boolean;
+  error?: boolean;
+  success?: boolean;
+  hint?: string;
+  rtl?: boolean;
+  formItem?: boolean;
+  icon?: React.ReactNode;
+  iconPosition?: IconPosition;
+};
+
+export const Input: React.FC<InputProps> = ({
+  id,
+  active,
+  compact,
+  disabled,
+  required,
+  error,
+  hint,
+  icon,
+  iconPosition,
+  label,
+  success,
+  value,
+  onFocus,
+  onBlur,
+  rtl,
+  formItem,
+  ...rest
+}) => {
+  const [focus, setFocus] = React.useState(false);
+  const wrapperClass = clsx('ray-text-field', {
+    'ray-text-field--compact': compact,
+    'ray-text-field--disabled': disabled,
+    'ray-text-field--required': required,
+    'ray-text-field--success': success,
+    'ray-text-field--error': error,
+    'ray-text-field--has-value': value,
+    'ray-text-field--active': active || focus,
+    'ray-text-field--with-icon-start':
+      iconPosition === IconPosition.Start && icon,
+    'ray-text-field--with-icon-end': iconPosition === IconPosition.End && icon,
+    'ray-text-field--with-prepend':
+      iconPosition === IconPosition.Prepend && icon
+  });
+  const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
```

Comment:
nit: you could reuse this type

Reply:
oo ok good to know ,  didn't get it, can you provide an example so I can see what you meant, please?